src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IConversionExpression.cs

```diff
@@ -0,0 +1,2991 @@
+﻿using System;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.Semantics;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests
+{
+    // Test list drawn from Microsoft.CodeAnalysis.CSharp.ConversionKind
+    public partial class IOperationTests : SemanticModelTestBase
```

Comment:
Looks like there is a bug with expression tree conversions where the type of the lambda in IOperation doesn't match the type in the semantic model. IMO, this looks like a bug in semantic model (or possibly the compiler, as a kind of `NoConversion` doesn't seem correct to me, especially given the failure mode of standard lambda->delegate mismatched conversions). Filed https://github.com/dotnet/roslyn/issues/20291 to track.